### PR TITLE
Fixed incorrect date cut-off time on forwarding events. Take timezone…

### PR DESCRIPTION
**Pull request by hazrulnizam**
… into account.

## Description

This fixes the visual bug on the DateLineView in the forwarding event page so that it cuts off at 12am according to the devices timezone, instead of cutting off at 12am UTC regardless.

## Motivation and Context

I wanted to make this change because as I am living in a GMT+8 timezone, the cut-off for 'Today', 'Yesterday', and previous dates happens at 8.00am instead of 12.00am in the current version.

## How Has This Been Tested?

I tested this by building the debug apk and running it on my device Samsung Galaxy S21.

I tested using my default GMT+8 timezone and also switching to a GMT-7 timezone to see the difference. 

While doing this, I came across another issue where if your timezone is in the negatives, the first DateLineView will show as 'Yesterday' instead of 'Today', and it will display another 'Yesterday' for the next cut-off making the page show 'Yesterday' two times. This has also been fixed in this PR.
## Screenshots (if appropriate):

Unable to take screenshots due to security policy.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [Contribution document](../docs/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by michaelWuensch** [comment]
Thank you so much!
I recently noticed it being off on my device for one hour as well. Yesterday I wanted to fix it, but didn't find the time. Abd now I see it is done! :)

I'll test it when I am back from work.

Furthermore I guess the transaction history has the same problem...

**Comment by hazrulnizam** [comment]
Apologies in advance if my code is not very nice. I am a newbie. Feel free to modify it as you see fit.

> Furthermore I guess the transaction history has the same problem...

I haven't looked at transaction history yet. At this point that page is not loading anything for me.

**Review by michaelWuensch — CHANGES_REQUESTED** [review]
I have added some comments in the review. 
If you can test these changes and confirm that it gives correct results, I would like you to change it before merging, as it should result in simpler code. :)

File: app/src/main/java/app/michaelwuensch/bitbanana/forwarding/listItems/DateItem.java [hunk 1]
```diff
@@ -12,7 +14,8 @@ public DateItem(long timestamp) {
         // To get the date line to show up at the correct position in the sorted list, we have to set its timestamp correctly.
         // We set it to 1 nanosecond before the day ends.
         // 86400000000000 = Nanoseconds of one day (60 * 60 * 24 * 1000 * 1000 * 1000)
-        mTimestampNS = (timestamp - (timestamp % 86400000000000L) + (86400000000000L - 1));
```
[the remark this answers is not in the document]
Comment (michaelWuensch) [reply app/src/main/java/app/michaelwuensch/bitbanana/forwarding/listItems/DateItem.java:15]: Actually I wanted to quote the next to lines as well, sorry, my bad :smile:

**Review by michaelWuensch — APPROVED** [review]

**Comment by hazrulnizam** [comment]
Hang on. I just found out my code does not 100% work as intended. Currently here at 5.57am, all forwards from 12.00am until now is shown above "Today" (not under any date headers), while all forwards that were yesterday are now shown under "Today", which is not intended behaviour. This is quite confusing.

I will cancel the pull request for now, and will re-request once I figured it out.

> I am still confused why the getYesterday() and getToday() functions need to be changed

I changed those functions to fix the "two Yesterdays" issue when the timezone is set to negative e.g. UTC-8 (Los Angeles). It works when I tested it yesterday but now I am not so sure. Let me close this PR first.

**Comment by michaelWuensch** [comment]
Okay, no hurry, I am quite busy untli the weekend anyway.
I actually saw that the DateLine on the transaction history is done quite differently with no complaints in the last years, so I suspect it is actually working. 
There I used SimpleDateFormat which already takes TimeZone into account.
Maybe looking at that helps.

**Comment by hazrulnizam** [comment]
You were right. I used the code from transaction history and used it on forwarding events and it now works beautifully. Thanks for the suggestion. New PR is here https://github.com/michaelWuensch/BitBanana/pull/15.

**Pull request closed without merging**